src/tsig/mod.rs

```diff
@@ -1316,11 +1316,12 @@ impl<'a, Octs: Octets + ?Sized> MessageTsig<'a, Octs> {
         let mut start = section.pos();
         let mut record = section.next()?;
         loop {
+            let record_start = section.pos();
             record = match section.next() {
```

Comment:
As far as I can tell, this doesn't seem to do the thing it says in the comment which is "checks that there is exactly one TSIG record in the additional section". Should it be doing that or should that be removed from the comment (or am I missing something)?

Reply:
True, it only checks that the last record is a TSIG record. I want to say that’s fine, but I’m honestly not sure.

Reply:
RFC 8945 states:
> If an incoming message contains a TSIG record, it MUST be the last record in the additional section. Multiple TSIG records are not allowed. If multiple TSIG records are detected or a TSIG record is present in any other position, the DNS message is dropped and a response with RCODE 1 (FORMERR) MUST be returned

Src: https://datatracker.ietf.org/doc/html/rfc8945#name-server-processing-of-reques

So it seems like something that's probably a good thing to do. But it also sounds like it should return `Err` in this case instead of `None`.

Reply:
Thanks for digging that up! So I guess that should indeed be fixed. Perhaps it’s better in a separate PR? (What with it changing the return type and all that.)

Reply:
Yes that's probably a good idea